Smartapi compliance with autogeneration

achave11-ucsc opened the pull request:
Please make sure that you've checked the following before submitting your pull request. Thanks!

- [] Check that you pass the basic style checks and unit tests by running `mvn clean install`

achave11-ucsc committed:
Made shell script an executable using `chmod`, followed by executing the script in `pom.xml`.
This simplifies the translation process from swagger to openapi, and gathers
the processes into a single file.

Modifies some tags in `Description.java`, updated version
and added termsOfService link to ga4gh standards and terms of service.

Regenerated `openapi` and `swagger` schemas to include the changes previously
mentioned, as well as making the `openapi` schemas compliant with smart-api.

`openapi` modifications where carried on by `swagger2open.sh` script which,
translated the swagger.json schemas to openapi style via web request. And
parsed de document by removing excess tags like `postion` and adding required
parameters like `server-url` and adding the `NIHdatacommons` tag.

achave11-ucsc committed:
Fixed issues with openapi parser. To include only one tag of NIH in the proper place.
Fixing script to not halt build process if there is no connection to the webservice api.

achave11-ucsc committed:
Removed compile step from maven build in pom file.

New version of openapi.yaml which deletes the NIHdatacommons tag that was
being excessively generated.

Reverted swagger to original, modified version and termsOfService manually to avoid
rebuild.

Fixed shell script to not fail if the curl request does not go through.

achave11-ucsc committed:
Modified script to now fail maven build if something goes wrong with
the curl request. Removed json swagger generator.

coverbeck commented on dockstore-webservice/src/main/java/io/dockstore/webservice/resources/Description.java:
```diff
@@ -40,7 +41,7 @@
 @SwaggerDefinition(info = @Info(description =
         "This describes the dockstore API, a webservice that manages pairs of Docker images and associated metadata such as "
-                + "CWL documents and Dockerfiles used to build those images", version = "1.3.0", title = "Dockstore API", contact = @Contact(name = "Dockstore@ga4gh", email = "theglobalalliance@genomicsandhealth.org", url = "https://github.com/ga4gh/dockstore"), license = @License(name = " GNU Lesser General Public License", url = "https://www.gnu.org/licenses/lgpl-3.0.en.html")), consumes = "app ...
+                + "CWL documents and Dockerfiles used to build those images", version = "1.4.1", title = "Dockstore API", termsOfService = "https://www.ga4gh.org/policies/termsandconditions.html", contact = @Contact(name = "Dockstore@ga4gh", email = "theglobalalliance@genomicsandhealth.org", url = "https://github.com/ga4gh/dockstore"), license = @License(name = " GNU Lesser General Public License ...
```
I commented somewhere else, but that is the wrong license. At least it's different than the Dockstore one.

coverbeck replied:
I also don't think we should be using the GA4GH terms and conditions.

Can you just leave it empty?

coverbeck commented on dockstore-webservice/src/main/resources/swagger2open.sh:
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# Using https://mermade.org.uk/openapi-converter, generate openapi schema sources.
+STATE=`curl -F "filename=@src/main/resources/swagger.yaml" https://mermade.org.uk/api/v1/convert -o src/main/resources/open_api.yaml -w "%{http_code}"`
```
I would call this HTTP_STATUS, or STATUS_CODE, to indicate that is holding an HTTP status code. STATE is confusing.

coverbeck commented on dockstore-webservice/src/main/resources/swagger2open.sh:
```diff
@@ -0,0 +1,22 @@
+  # Clean directory from excess files.
+  rm src/main/resources/open_api.yaml src/main/resources/openapi1.yaml src/main/resources/openapi2.yaml
+else
+  echo "URL not available, graciously terminate"
```
You don't know why it failed, so I wouldn't say `URL not available`. I would have a more generic error message, like `Error converting swagger to openapi; skipping openapi generation`

coverbeck commented on dockstore-webservice/src/main/resources/swagger2open.sh:
```diff
@@ -0,0 +1,22 @@
+# Using https://mermade.org.uk/openapi-converter, generate openapi schema sources.
+STATE=`curl -F "filename=@src/main/resources/swagger.yaml" https://mermade.org.uk/api/v1/convert -o src/main/resources/open_api.yaml -w "%{http_code}"`
+eval "$STATE"
+if [ \( "$?" = 0 \) -o \(  "$STATE" = "200" \) ]; then
```
Does this work? I don't know bash very well, so maybe it does. But did you test:

1. Works
2. Fails because curl fails, e.g., the file is missing
3. Fails because the call returns a non-200 value.

achave11-ucsc replied:
Basically what I am doing is, executing the command defined by the variable. The block is only true if the curl request has an exit code of 0, and if the response to the http request is a 200. Otherwise it quits with a 0 exit code. 

Methods for testing:
* Swap 200 for a 400 in the logic block, [gave error message] only 200 accepted.
* Swap 0 in exit code of command for 1, [gave error message] only 0 accepted.
* Changed url path to something different, [gave error message] 
* Changed file name, something not there, [gave error message] 

This satisfies the possible conditions of error, and only succeeds if both conditions are satisfied.

achave11-ucsc committed:
Removed termsOfService tag definition from Description.java
Did to for the swagger schemas file.

Rename variable to identify request status code.

Renamed failed status massage to a more generic one.

achave11-ucsc committed:
Fixed logic for proceeding with file parsing if the curl execution was
successful.

achave11-ucsc committed:
Reverted termsOfService tag
